commands.js

```diff
@@ -1465,6 +1465,26 @@ var commands = exports.commands = {
 	},
 	forcerenamehelp: ["/forcerename OR /fr [username], [reason] - Forcibly change a user's name and shows them the [reason]. Requires: % @ & ~"],
 
+	hidetext: function (target, room, user) {
+		if (!target) return this.parse('/help hidetext');
+
+		var reason = this.splitTarget(target);
+		var targetUser = this.targetUser;
+		var name = this.targetUsername;
+		if (!targetUser) return this.errorReply("User '" + name + "' does not exist.");
+		if (!this.can('lock', targetUser) && !this.can('ban', targetUser, room)) return false;
+
+		if (!targetUser.locked && !Users.checkBanned(targetUser.latestIp)) {
+			if (room.bannedUsers[toId(name)] && room.bannedIps[targetUser.latestIp]) {
+				return this.errorReply("/hidetext does not yet support removing messages from users banned from rooms.");
```

Comment:
I'm trying to figure out how to get the client to clear messages from one room only. I'm working on it but if you want the whole feature in one go I'll need some more time to patch the client up properly.